### ISSUE TITLE
Relative homology

### ISSUE THREAD

**Issue by mglisse** [GUDHI/gudhi-devel#121]
Since several users have asked: we should provide an example showing how to use persistence to compute the relative homology of a complex and a subcomplex (mostly how to interpret the result).

**Comment by MathieuCarriere** [comment]
I agree. Extended persistence could also be interesting since I have received a few messages from people trying to use Gudhi for computing extended persistence.

**Comment by VincentRouvreau** [comment]
Was this issue fixed by #215 ? @mglisse @MathieuCarriere

**Comment by mglisse** [comment]
I don't think #215 contains an example showing how to compute the relative homology of a pair (L,K), so while it is related, it answers a different question.

**Comment by mglisse** [comment]
In a discussion, we mentioned the possibility of adding an API `relative_betti_numbers(x,y)` to compute the relative homology of `(K_y, K_x)`.

**Comment by shilangyu** [comment]
Has there been any progress in this space? If not, what are the possible approaches to compute relative homology of a complex and its subcomplex using the current API? I'd be happy to contribute documentation upon some guidance.

**Comment by OnDraganov** [comment]
Coning is an easy alternative. It increases the complex size, but the result can be interpreted as-is.

If we have a pair (K,L), `st` is the `SimplicialTree` representing K, and `subcomplex` is the list of the simplices in L, we can just do the following: 
```
new_vertex = len(points)  # gudhi uses vertices 0, ..., len(points) - 1
st.insert([new_vertex], 0)
for simplex in subcomplex:
    st.insert(simplex + [new_vertex], st.filtration(simplex))
st.compute_persistence()
relative_pd = st.persistence()
```
